lib/Target/DebugInfo/EmitHGLDD.cpp

```diff
@@ -641,45 +641,22 @@ EmittedExpr FileEmitter::emitExpression(Value value) {
   auto result = cast<OpResult>(value);
   auto *op = result.getOwner();
 
-  // If the operation has only this one result and is named in some form, use
-  // that name.
-  if (op->getNumResults() == 1) {
-    // If a `hw.verilogName` is available, emit the value as just a reference to
-    // that name.
+  // If this is a combinational operation, and it produces a single result, and
+  // it has a `hw.verilogName` associated with it, use that name.
+  if (op->getNumResults() == 1 && isa<comb::CombDialect>(op->getDialect()))
     if (auto name = op->getAttrOfType<StringAttr>("hw.verilogName");
         name && !name.empty())
       return {hglddSigName(name), result.getType()};
```

Comment:
This might be dead code. IIRC comb operations don't get hw.verilogName attributes.

Reply:
Great point!